plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -0,0 +1,775 @@
+                              DockerConnectionFactory connectionFactory,
+                              DockerRegistryAuthResolver authResolver,
+                              KubernetesLabelConverter kubernetesLabelConverter,
+                              KubernetesEnvVar kubernetesEnvVar,
+                              KubernetesContainer kubernetesContainer,
+                              KubernetesService kubernetesService,
+                              DockerApiVersionPathPrefixProvider dockerApiVersionPathPrefixProvider,
+                              @Named("che.openshift.endpoint") String openShiftApiEndpoint,
+                              @Named("che.openshift.username") String openShiftUserName,
+                              @Named("che.openshift.password") String openShiftUserPassword,
+                              @Named("che.openshift.project") String openShiftCheProjectName,
+                              @Named("che.openshift.serviceaccountname") String openShiftCheServiceAccount) {
+        super(connectorConfiguration, connectionFactory, authResolver, dockerApiVersionPathPrefixProvider);
+        this.openShiftCheProjectName = openShiftCheProjectName;
+        this.openShiftCheServiceAccount = openShiftCheServiceAccount;
+        this.kubernetesLabelConverter = kubernetesLabelConverter;
+        this.kubernetesEnvVar = kubernetesEnvVar;
+        this.kubernetesContainer = kubernetesContainer;
+        this.kubernetesService = kubernetesService;
+
+        Config config = new ConfigBuilder().withMasterUrl(openShiftApiEndpoint)
+                .withUsername(openShiftUserName)
+                .withPassword(openShiftUserPassword).build();
+        this.openShiftClient = new DefaultOpenShiftClient(config);
+    }
+
+    /**
+     * @param createContainerParams
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public ContainerCreated createContainer(CreateContainerParams createContainerParams) throws IOException {
+        String containerName = getNormalizedContainerName(createContainerParams);
+        String workspaceID = getCheWorkspaceId(createContainerParams);
+        // Generate workspaceID if CHE_WORKSPACE_ID env var does not exist
+        workspaceID = workspaceID.isEmpty() ? generateWorkspaceID() : workspaceID;
+        String imageName = createContainerParams.getContainerConfig().getImage();
+
+        Set<String> containerExposedPorts = createContainerParams.getContainerConfig().getExposedPorts().keySet();
+        Set<String> imageExposedPorts = inspectImage(imageName).getConfig().getExposedPorts().keySet();
+        Set<String> exposedPorts = getExposedPorts(containerExposedPorts, imageExposedPorts);
+
+        boolean runContainerAsRoot = runContainerAsRoot(imageName);
+
+        String[] envVariables = createContainerParams.getContainerConfig().getEnv();
+        String[] volumes = createContainerParams.getContainerConfig().getHostConfig().getBinds();
```

Comment:
@garagatyi yeah we providing just basic volume support in this PR. 

@snjeza is working on a PR to improve the volumes usage of the OpenShiftConnector and make use of [Kubernetes Persistent Volumes](https://kubernetes.io/docs/user-guide/persistent-volumes/) instead of current hostpath volumes. That will significantly change this code so I guess it does not really make sense to address it here. 

By the way to make sure we are able to test it correctly how can we manage to have a container in Che that doesn't have bind-mounted volumes but just regular volumes? Using a docker-compose custom recipe?